org.adempiere.ui.zk/WEB-INF/src/web/js/dragdrop/attachment/drag-drop-attachment.js

```diff
@@ -0,0 +1,199 @@
+ * 
+ * @author Logilite Technologies
+ * @since October 16, 2017
+ */
+function DropToAttachFiles() {
+
+    var clsName = " attachment-drag-entered";
+
+    this.init = init;
+    this.findAndReplace = findAndReplace;
+
+    function init(formID, dropBtnID, desktopID, progBarID, progLabelID, maxUploadSize, sessionid) {
+
+        var seqID = 0;
+        var highlightdragging;
+        var dropBtn = document.getElementById(dropBtnID);
+        var progressbar = document.getElementById(progBarID);
+
+        var isSupported = {
+            dnd: 'draggable' in document.createElement('span'),
+            formdata: !!window.FormData,
+            progress: "upload" in new XMLHttpRequest
+        };
+
+        if (isSupported.dnd) {
+            dropBtn.ondragover = function() {
+                return false;
+            };
+
+            dropBtn.ondragenter = function(e) {
+                highlightdragging = e.target;
+                this.className = this.className + clsName;
+                return false;
+            };
+
+            dropBtn.ondragleave = function(e) {
+                if (e.target === highlightdragging)
+                    this.className = findAndReplace(this.className, clsName, '');
+                return false;
+            };
+
+            dropBtn.ondrop = function(e) {
+                e.preventDefault();
+
+                var requestSize = 0;
+                var files = e.dataTransfer.files;
+                for (var i = 0; i < files.length; i++) {
+                    requestSize = requestSize + files[i].size;
+                }
+
+                requestSize = Math.round(requestSize / 1024);
+
+                if (maxUploadSize && requestSize > maxUploadSize) {
+                    alert('The request was rejected because its size (' + requestSize + ') exceeds the configured maximum (' + maxUploadSize + ')');
+                    return;
+                }
+
+                progressbar.style = 'display: block';
+                progressbar.parentNode.style = 'display: block';
+                progressBarUploadStatus(progBarID, progLabelID, 0, 'Starting Upload...');
+                this.className = findAndReplace(this.className, clsName, '');
+
+                uploadFiles(files, seqID++);
+                return false;
+            };
+        } else {
+            clsName = "";
+            alert("Click to Upload Files");
+        } // drop listener
+
+        function uploadFiles(files, sid) {
+            var formData = isSupported.formdata ? new FormData() : null;
+            if (isSupported.formdata) {
+                for (var i = 0; i < files.length; i++) {
+                    formData.append('file', files[i], files[i].name);
+                }
+            }
+
+            var xhr = new XMLHttpRequest();
+            var url =
+                "/webui/zkau/upload;jsessionid=" + sessionid + "?uuid=" + formID + "&dtid=" + desktopID + "&sid=" + sid + "&native=true";
```

Comment:
adding jsessionid to url is a security risk. this is usually not necessary since xhr request should be using the same servlet session by default.